src/ray/protobuf/gcs.proto

```diff
@@ -105,6 +105,10 @@ message ActorTableData {
   uint64 max_reconstructions = 7;
   // Remaining number of reconstructions.
   uint64 remaining_reconstructions = 8;
+  // The IP address of the node that the actor is running on.
+  string ip_address = 9;
+  // The port that the actor is listening on.
+  int32 port = 10;
```

Comment:
How are these ports assigned? Is there a way to configure a port range to use?

Reply:
The port is automatically assigned by GRPC server. It currently doesn't seem to provide a way to specify a port range, while there are some people tried to loop over a range of ports manually with `AddListeningPort`, it was reported that this function might succeed even if a port is already bound, I haven't test myself though. 

https://github.com/grpc/grpc/blob/master/include/grpcpp/server_builder_impl.h#L125
https://groups.google.com/forum/#!topic/grpc-io/zOgTWQnMO6s

Reply:
Hm ok, this is probably fine for prototyping but users will expect to be able to set a port range in production. This can be a flag config passed to raylet and it can assign them sequentially to workers, or something like that.

Reply:
By the way, currently there is no health checking of these workers from the raylet right? It would probably be a good idea for the raylet to kill actors that are unresponsive to checks (due to invalid port, internal errors, etc).

Reply:
The purpose for specifying worker port range is to configure firewall/nat stuff in public cloud or similar scenarios? (In company private networks this is mostly not an issue).

Reply:
for the health checking, the PR to convert raylet client to grpc implements worker to raylet heartbeat, this PR is close to merge, it should be very easy to add killing actors on top of this.